### ISSUE TITLE
RLTS - Configurable Multi-Batching

### ISSUE THREAD

**Issue by jeremyk-91** [palantir/atlasdb#2086]
#2083 implements a smart-batching timestamp service, but clients can only have one open batch at a time. 

We can probably improve performance further by having a finite number of batches at a time. If we have N batches then if we can space them RTT/N apart we could probably see even better performance. We can maintain some kind of RTT estimate and use that to determine how long to hold off before releasing a batch.

**Comment by jeremyk-91** [comment]
There's also a random idea (pun intended), where we run N RLTSs and dispatch to a random one, to achieve N parallelism. Probably a ton simpler, though we might have less control over the distribution.

**Comment by nziebart** [comment]
Yeah I was thinking about the random option also. I'm not sure how precise we can get in terms of scheduling below 1ms.
